### PR TITLE
reorganised ByteArray, Bytes and test cases

### PR DESCRIPTION
**Pull request by kartik-karz**

### CODE REVIEW

File: python/common/org/python/types/ByteArray.java [hunk 1]
```diff
@@ -776,7 +776,27 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
             __doc__ = "B.isdigit() -> bool\n\nReturn True if all characters in B are digits\nand there is at least one character in B, False otherwise."
     )
     public org.python.Object isdigit(java.util.List<org.python.Object> args, java.util.Map<java.lang.String, org.python.Object> kwargs, java.util.List<org.python.Object> default_args, java.util.Map<java.lang.String, org.python.Object> default_kwargs) {
```
Comment (eliasdorneles) [inline python/common/org/python/types/ByteArray.java]: This is already implemented in `Bytes` class, you should reuse that method instead of rewriting it.

Comment (kartik-karz) [reply python/common/org/python/types/ByteArray.java]: Oh...should I copypasta it or call it?

Comment (eliasdorneles) [reply python/common/org/python/types/ByteArray.java]: Call it.

Comment (kartik-karz) [reply python/common/org/python/types/ByteArray.java]: shall I create a utility method in Bytes class like _isalpha and change that in isalpha as done previously? just for consistency... is there any other significant reason for not doing that over there in Bytes?

Comment (eliasdorneles) [reply python/common/org/python/types/ByteArray.java]: > shall I create a utility method in Bytes class like _isalpha and change that in isalpha as done previously

Sounds good.

**Comment by kartik-karz** [comment]
Done! please check

**Comment by eliasdorneles** [comment]
Hi @kartik-karz, I appreciate that you are interested in contributing, but it's really important that you test the changes you make.
Things that are not tested add a lot to the burden of people that are already overburdened -- it's not very polite to ask someone to check something that you have not done a minimal testing.

This needs adding tests and updating the method arguments (it doesn't work at all without [the correct signature](http://voc.readthedocs.io/en/latest/reference/signatures.html)).
You will also want to check the style check failures that the build result is showing.

**Comment by kartik-karz** [comment]
Hey, I guess it's time for some celebration... unless I forgot something else...

**Comment by goanpeca** [comment]
>unless I forgot something else...

Wait for all tests to pass

**Comment by sziraqui** [comment]
Sorry my commit will probably cause a conflict, didn't see you are also working on the same method.

**Comment by kartik-karz** [comment]
@eliasdorneles is there a problem with the code? I have tried almost all the test cases manually as well as with the test cases but CIS test cases show pending for three tests and doesn't even show the logs...

**Comment by freakboy3742** [comment]
@kartik-karz The build errored out; I've just manually restarted it. Apologies for the confusion caused.

File: docs/how-to/contribute.rst [hunk 2]
```diff
@@ -116,6 +116,19 @@ Or you can use Cricket, a GUI tool for running test suites. To start cricket in
...
+Hopefully you are all setup by now and would like to find some issues to work on, you can always go
+to the voc repo and find some issues to work on or go to  python/common/org/python/types and browse through the 
+files over there to find some unimplemented methods and implement it.
+These are the files which are required by the voc to convert the python code into java bytecode, 
+after you have implemented those you will have to go to tests/builtins and find the python file for the 
+java method you modified (in /python/types) and write a few test cases to check if it's working properly.
+You can also test the code like you did with your hello world program :doc:`/tutorial/tutorial-0` and
+try out your self written methods and watch them transform into java class with the magic of voc.
+If you feel that it's working then submit a PR and wait till your code gets merged.
+
```
Comment (freakboy3742) [inline docs/how-to/contribute.rst]: This change is completely unrelated to the isdigit check.

**Comment by kartik-karz** [comment]
@freakboy3742 no problem XD I'm just happy I don't have to create a separate env in my system with Py3.4 for and run the full test again( took me ages), I have also removed the subtopic from the docs.
I just felt that there should have been a place to look so that newbies can easily find out places to contribute... I had to ask around the chat and look through the code to find the stuff I was looking for and felt that others could save their time by going there directly.
Anyway, as you suggested I will try do a PR later for the docs if I find anything else to be helpful.

**Comment by kartik-karz** [comment]
I have learnt a life lesson which I think I will never forget... never use Github merge conflict tool to resolve conflicts...also always mention what all files you are working on before so that conflict doesn't happen at all in the first place.

**Comment by cflee** [comment]
@kartik-karz Can you run `flake8` and `ant checkstyle` locally, while you're in the `voc` project folder? I think the style checks are not working properly on the CI server, so please run the Java `ant checkstyle` locally and ensure that it doesn't have any style issues. That way we don't have to leave a ton of minor comments about indentation, whitespace and stuff like that. Thanks!

**Comment by kartik-karz** [comment]
@cflee Okay I have fixed all the errors except don't use bare except in utils.py which was there in master already. Also, can you please check if the CIS has bugged out again? one of the tests has been going on for a while now (~15mins ) even after 100% completion which normally takes (~10mins)

**Comment by kartik-karz** [comment]
@freakboy3742 well, I had changed the title... and before you ask what a test_float case is doing here, it was not passing flake8 and needed an extra newline... see I have properly communicated this time...can I get this merged?? :neutral_face:

**Comment by kartik-karz** [comment]
It's done on the newer PR and thus this PR is useless
